src/PointwiseFunctions/Hydro/EquationsOfState/RegisterDerivedWithCharm.cpp

```diff
@@ -0,0 +1,39 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <cstddef>
+
+#include "Parallel/RegisterDerivedClassesWithCharm.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/Factory.hpp"
+namespace EquationsOfState {
+
+struct RegisterClassesFunctor {
+  RegisterClassesFunctor() = default;
+  RegisterClassesFunctor(const RegisterClassesFunctor&) = delete;
+  RegisterClassesFunctor(RegisterClassesFunctor&&) = default;
+  RegisterClassesFunctor& operator=(const RegisterClassesFunctor&) = delete;
+  RegisterClassesFunctor& operator=(RegisterClassesFunctor&&) = default;
+
+  template <typename T>
+  void operator()(T /*t*/) {
+    using type = tmpl::type_from<T>;
+    Parallel::register_classes_with_charm<type>();
+  }
+};
+template <bool IsRelativistic, size_t ThermodynamicDim>
+void register_derived_subset_with_charm() {
+  using derived_equations_of_state =
+      typename EquationsOfState::detail::DerivedClasses<IsRelativistic,
+                                                        ThermodynamicDim>::type;
+  tmpl::for_each<derived_equations_of_state>(RegisterClassesFunctor{});
```

Comment:
could also do `Register...` -> 
```cpp
[](auto type_v) { 
    using type = tmpl::type_from<decltype(type_v)>;
    Parallel::register_classes_with_charm<type>();
}
```
Less code and is usually what we do :)

Reply:
Or even `Parallel::register_classes_with_charm(derived_equations_of_state{});`

Reply:
Ah this is nice!